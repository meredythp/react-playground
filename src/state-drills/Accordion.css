button {
  padding: 3px;
  margin: 6px;
  border: 1px solid black;
  background-color: lightcoral;
}

ul {
  text-align: left;
  list-style: none;
}

.content {
  padding: 0 0 0 9px;
}

.none {
  display: none;
}

.block {
  display: block;
}