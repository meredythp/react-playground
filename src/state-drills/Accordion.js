import React from 'react';
import './Accordion.css'

class Accordion extends React.Component {
  static defaultProps = {
    sections: [],
  }

  state = {
    currentSectionIndex: null,
  };

  HandleButtonClick(index) {
    this.setState({
      currentSectionIndex: index,
    })    
  }

  /* DisplayContent(index) {
    let classState = index === this.state.currentSectionIndex ? "block" : "none";
    return classState
  } */

  Item(section, index) {
    return (
      <li key={index}>
        <button onClick={() => this.HandleButtonClick(index)}>
            {section.title}
        </button>
        {/* <div id="content" className={this.DisplayContent(index)}> */}
        {this.props.sections.length > 0 && index === this.state.currentSectionIndex && <div className="content"><p>{section.content}</p></div>}
        {/* </div> */}
      </li>
    )
  }

  render() {
    return (
      <div className="section-wrapper">
        <ul className="main-ul">
          {this.props.sections.map((section, index) => (
            this.Item(section, index)
          ))}
        </ul>
      </div>
    )
  }
}

export default Accordion;