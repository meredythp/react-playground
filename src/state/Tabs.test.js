import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Tabs';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({
  adapter: new Adapter()
});

describe.skip(`Tabs Component`, () => {
  const tabsProp = [
    {
      name: 'First tab',
      content: 'Lorem ipsum dolor sit...'
    },
    {
      name: 'Second tab',
      content: 'Laboriosam exercitationem quos...'
    },
    {
      name: 'Third tab',
      content: 'Fugit, sapiente aspernatur...'
    },
  ];

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the first tab by default', () => {
    const tree = renderer
      .create(<Accordion tabs={tabsProp} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})