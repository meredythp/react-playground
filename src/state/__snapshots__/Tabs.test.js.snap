// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Component closes the first tab and opens any clicked tab 1`] = `
<div
  className="button-div"
>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Third tab
  </button>
  <div
    className="content"
  >
    Laboriosam exercitationem quos...
  </div>
</div>
`;

exports[`Tab Component renders empty given no tabs 1`] = `
<div
  className="button-div"
/>
`;

exports[`Tab Component renders the first tab by default 1`] = `
<div
  className="button-div"
>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Third tab
  </button>
  <div
    className="content"
  >
    Lorem ipsum dolor sit...
  </div>
</div>
`;
